src/types/shell/xdg_shell_v6.rs

```diff
@@ -181,6 +183,21 @@ impl XdgV6ShellSurface {
         }
     }
 
+    pub fn for_each_surface(&self, iterator: &FnMut(SurfaceHandle, i32, i32)) {
+        unsafe {
+            extern fn c_iterator(wlr_surface: *mut wlr_surface, sx: i32, sy: i32, data: *mut c_void) {
+                unsafe {
+                    let iterator = &mut *(data as *mut Box<&Fn(SurfaceHandle, i32, i32)>);
+                    let surface = SurfaceHandle::from_ptr(wlr_surface);
+                    iterator(surface, sx, sy);
+                }
+            }
+            let mut iterator = Box::new(iterator);
```

Comment:
Does this need to be boxed? Can you not just pass the `&FnMut` as a function pointer to the function?